numpy/core/src/multiarray/datetime.c

```diff
@@ -2832,141 +2832,6 @@ convert_pyobject_to_timedelta(PyArray_DatetimeMetaData *meta, PyObject *obj,
-     */
-    if (dts.year < 1 || dts.year > 9999 || dts.sec == 60) {
-        return PyLong_FromLongLong(dt);
-    }
-
-    /* If the type's precision is greater than days, return a datetime */
-    if (meta->base > NPY_FR_D) {
-        ret = PyDateTime_FromDateAndTime(dts.year, dts.month, dts.day,
-                                dts.hour, dts.min, dts.sec, dts.us);
-    }
-    /* Otherwise return a date */
-    else {
-        ret = PyDate_FromDate(dts.year, dts.month, dts.day);
-    }
-
-    return ret;
-}
-
-/*
- * Converts a timedelta into a PyObject *.
- *
- * Not-a-time is returned as the string "NaT".
- * For microseconds or coarser, returns a datetime.timedelta.
- * For units finer than microseconds, returns an integer.
- */
-NPY_NO_EXPORT PyObject *
-convert_timedelta_to_pyobject(npy_timedelta td, PyArray_DatetimeMetaData *meta)
-{
-    npy_timedelta value;
-    int days = 0, seconds = 0, useconds = 0;
-
-    /*
-     * Convert NaT (not-a-time) into None.
-     */
-    if (td == NPY_DATETIME_NAT) {
-        Py_RETURN_NONE;
-    }
-
-    /*
-     * If the type's precision is greater than microseconds, is
-     * Y/M/B (nonlinear units), or is generic units, return an int
-     */
-    if (meta->base > NPY_FR_us ||
-                    meta->base == NPY_FR_Y ||
-                    meta->base == NPY_FR_M ||
-                    meta->base == NPY_FR_GENERIC) {
-        return PyLong_FromLongLong(td);
-    }
-
-    value = td;
-
-    /* Apply the unit multiplier (TODO: overflow treatment...) */
-    value *= meta->num;
-
-    /* Convert to days/seconds/useconds */
-    switch (meta->base) {
-        case NPY_FR_W:
-            days = value * 7;
-            break;
-        case NPY_FR_D:
-            days = value;
-            break;
-        case NPY_FR_h:
-            days = extract_unit_64(&value, 24ULL);
-            seconds = value*60*60;
-            break;
-        case NPY_FR_m:
-            days = extract_unit_64(&value, 60ULL*24);
-            seconds = value*60;
-            break;
-        case NPY_FR_s:
-            days = extract_unit_64(&value, 60ULL*60*24);
-            seconds = value;
-            break;
-        case NPY_FR_ms:
-            days     = extract_unit_64(&value, 1000ULL*60*60*24);
-            seconds  = extract_unit_64(&value, 1000ULL);
-            useconds = value*1000;
-            break;
-        case NPY_FR_us:
-            days     = extract_unit_64(&value, 1000ULL*1000*60*60*24);
-            seconds  = extract_unit_64(&value, 1000ULL*1000);
-            useconds = value;
-            break;
-        default:
-            // unreachable, handled by the `if` above
-            assert(NPY_FALSE);
-            break;
-    }
-    /*
-     * If it would overflow the datetime.timedelta days, return a raw int
-     */
-    if (days < -999999999 || days > 999999999) {
-        return PyLong_FromLongLong(td);
-    }
-    else {
-        return PyDelta_FromDSU(days, seconds, useconds);
-    }
-}
```

Comment:
Maybe one should keep this?  Probably not as `.item()` but for some kind of way to convert to Python datetime?